presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeGracefulShutdown.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.spi.NodeState;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import org.testng.annotations.Test;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestPrestoNativeGracefulShutdown
+{
+    @Test
+    public void testGracefulShutdown() throws Exception
+    {
+        QueryRunner queryRunner = PrestoNativeQueryRunnerUtils.createNativeQueryRunner(true);
+        DistributedQueryRunner distributedQueryRunner = (DistributedQueryRunner) queryRunner;
+
+        int responseCode = distributedQueryRunner.sendWorkerShutdown(0,"INVALID_BODY");
+        assertEquals(responseCode, 400, "Expected a 400 Bad Request response for invalid body");
+
+        responseCode = distributedQueryRunner.sendWorkerShutdown(0,"");
+        assertEquals(responseCode, 400, "Expected a 400 Bad Request response for empty body");
+
+        responseCode =distributedQueryRunner.sendWorkerShutdown(0,"\"SHUTTING_DOWN\"");
```

Comment:
Can we add a bit more elaborate test ? Like waiting for some more time post this request to check that the worker has infact shutdown.

What are the e2e use-cases of this functionality ? Might be worth having an e2e test for those.

Reply:
> What are the e2e use-cases of this functionality ? Might be worth having an e2e test for those.

The graceful shutdown feature in Presto is used by other apps on top of Presto such as auto-scaling. I think ensuring the node goes down is sufficient for the Prestissimo E2E tests.
@anandamideShakyan can you check what tests exist on the Java side?